SimG4CMS/Forward/src/ZdcSD.cc

```diff
@@ -416,14 +403,13 @@ void ZdcSD::setNumberingScheme(ZdcNumberingScheme* scheme) {
   if (scheme != nullptr) {
     edm::LogInfo("ForwardSim") << "ZdcSD: updates numbering scheme for " 
 			       << GetName();
-    if (numberingScheme) delete numberingScheme;
+    delete numberingScheme;
     numberingScheme = scheme;
```

Comment:
use unique_ptr
maybe all those commonality among call SD should be consolidated into a (template?) base class